db/filename_test.cc

```diff
@@ -171,6 +171,68 @@ TEST_F(FileNameTest, Construction) {
   ASSERT_EQ(kMetaDatabase, type);
 }
 
+TEST_F(FileNameTest, NormalizePath) {
+
+  std::string separator(1, kFilePathSeparator);
+  std::stringstream given, toNormalize;
+  std::string noeLeadingSlash, twoChars1, twoChars2,slashOnly, uncOnly;
+
+
+  given << separator << separator << "SERVER" << separator << "fileName";
+  toNormalize << separator << "pathel" << separator << "pathel_1";
+
+  std::string expected = given.str();
+
+  ASSERT_EQ(expected, NormalizePath(given.str()));
+
+  expected = toNormalize.str();
+
+  ASSERT_EQ(expected, NormalizePath(toNormalize.str()));
+
+  // No leading slash
+  toNormalize << "folder" << separator << "fileName.ext";
+  given << "folder" << separator << "filename.ext";
```

Comment:
These do have leading slashes (or two).  You are appending to the std::stream so the name is //SERVER/path/folder/filename.ext, not just "folder/fileName.ext".

If you change to not use std::stream, it will become clearer what you are trying to do.